src/mpl/include/mpl_gpu.h

```diff
@@ -29,6 +29,7 @@ typedef enum {
 typedef struct {
     MPL_pointer_type_t type;
     MPL_gpu_device_handle_t device;
+    MPL_gpu_device_attr attr;
```

Comment:
My only issue is the name of the struct member. Maybe `device_attr` or `dev_attr`?

Reply:
Sounds good.

An unrelated side note: it appears to me that at least cuda does its internal query cache/optimization already. We need measure the overhead of passing info and without passing info to yaksa pack/unpack. Maybe it not necessary to do this opmization.

Reply:
Fixed. Renamed to `device_attr`.